### ISSUE TITLE
feat: allow to use names instead of site_id

### ISSUE THREAD

**Issue by moltar** [netlify/cli#383]
**- Do you want to request a *feature* or report a *bug*?**

feature

**- What is the current behavior?**

All commands require exact `site_id` value, which is seems to be a `UUID`.

**- What is the expected behavior?**

Allow to use the site name (slug) as the value.

The use case for this is CI.

I am creating a "review app" in GitLab, which uses an auto-generated name.

But to delete the review app, it requires a unique ID.

**- Please mention your node.js, and operating system version.**

N/A

**Comment by DavidWells** [comment]
Hello @moltar once this https://github.com/netlify/cli/pull/384 is merged you will be able to do:

```
netlify status --json | jq '.siteData."site-id"'
// your id 
```

You can also call the new `api`, once https://github.com/netlify/cli/pull/379 is merged, command like so:

```
netlify api listSites --data '{ "name": "your-site-name"}' | jq '.[0].id'
```

I'm using https://stedolan.github.io/jq/ to parse the JSON returned from the command.

**Comment by DavidWells** [comment]
One of the reasons the CLI commands use `site-id` is to avoid the more expensive lookup of the sites by name via `listSites` API method. (this will scan all sites in your account)

**Comment by moltar** [comment]
Yes, I am using `jq` as well:

```sh
SITE_ID = $(netlify sites:list --json | jq ".[] | select(.name == \"$CI_BUILD_REF_SLUG\")" | jq '.site_id' -r)
```

So it does save some characters, but it's not ultimately what the issue is about :)

This requires a dependency on `jq`.

I think it should work like:

```sh
## create already works
netlify sites:create --name $CI_BUILD_REF_SLUG

## now I want to delete the same way
netlify sites:delete --force --name $CI_BUILD_REF_SLUG
```

**Comment by erezrokah** [comment]
Closing in favour of https://github.com/netlify/cli/issues/157
